OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ChildController.cs

```diff
@@ -101,6 +108,59 @@ public async Task<IActionResult> GetUsersChildById(Guid id)
             return Ok(await service.GetByIdAndUserId(id, userId).ConfigureAwait(false));
         }
 
+        /// <summary>
+        /// Get approved children from the database by workshop id.
+        /// </summary>
+        /// <param name="workshopId">Id of the parent.</param>
+        /// <param name="offsetFilter">Filter to get a part of all children that were found.</param>
+        /// <returns>The result is a <see cref="SearchResult{ChildDto}"/> that contains the count of all found children and a list of children that were received.</returns>
+        [HasPermission(Permissions.WorkshopEdit)]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(SearchResult<ChildDto>))]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        [HttpGet("{workshopId}")]
+        public async Task<IActionResult> GetApprovedByWorkshopId(Guid workshopId, [FromQuery] OffsetFilter offsetFilter)
+        {
+            var userHasRights = await this.IsUserProvidersOwnerOrAdmin(workshopId).ConfigureAwait(false);
+            if (!userHasRights)
+            {
+                return StatusCode(403, "Forbidden for another providers.");
+            }
+
+            return Ok(await service.GetApprovedByWorkshopId(workshopId, offsetFilter).ConfigureAwait(false));
+        }
+
+        private async Task<bool> IsUserProvidersOwnerOrAdmin(Guid workshopId)
```

Comment:
(Code style readability concern) 
1. If `User` is not in `provider` role, you return `true` for method named `IsUserProvidersOwnerOrAdmin`. I'm bit concerned. Either change method name or store result of check (`User.IsInRole(nameof(Role.Provider).ToLower())`) into variable with meaningful name
2. If `User` is not in `provider` role, then there is no sense to call `providerService.GetProviderIdForWorkshopById()`, you are not using its result for this case. So, you could call it later inside `try`/`catch` after `providerService.GetByUserId()`
3. You have single `return true` after `if (User.IsInRole(...)) { ... }` with more than 20 lines of code. Isn't it better to invert condition in mentioned `if` and have small one-line `return true`? You'll definitely improve your code readability.
4. Code like 
```
if (providerId != provider.Id)
{
    return false;
} 
```
with subsequent `return true` looks like long way to write simple `return providerId == provider.Id;`
5. Same for `if (!isUserRelatedAdmin)`
6. Inconsistent usage of `.ConfigureAwait(false)`, it is not used for `providerService.GetProviderIdForWorkshopById()`. Is there any reason?